src/Qir/check-sources-formatted.ps1

```diff
@@ -0,0 +1,31 @@
+#Requires -PSEdition Core
+#Requires -Version 7.1
+
+param (
+    [Parameter()]
+    [String]
+    $DirPath = "$PSScriptRoot"
+)
+
+$tmpFile = "format.log"
+
+$OldErrorActionPreference = $ErrorActionPreference
+$ErrorActionPreference='Continue'
+"$DirPath/*.cpp","$DirPath/*.c","$DirPath/*.h","$DirPath/*.hpp" | get-childitem -Recurse | %{$_.fullname -replace "\\", "/"} `
+    | ?{$_ -notlike "*/Externals/*"} | ?{$_ -notlike "*/drops/*"} | ?{$_ -notlike "*/bin/*"} `
+    | %{clang-format -n -style=file $_} 2>format.log
+$ErrorActionPreference=$OldErrorActionPreference
+
+$filesRequireFormatting = get-content $tmpFile | ?{$_ -like "*: warning:*"} `
+                            | %{[string]::join(":",($_.split("warning:")[0].split(":") | select -SkipLast 3))} `
+                            | sort | unique
+Remove-Item $tmpFile
+
+if (! ("$filesRequireFormatting" -eq ""))
```

Comment:
```suggestion
if (-not ("$filesRequireFormatting" -eq ""))
```

Reply:
As an alternative:
```suggestion
if ("$filesRequireFormatting" -ne "")
```

Reply:
Just curiosity:
May I know the advantages of `-not` over `!`?
In [about_Logical_Operators](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_logical_operators?view=powershell-7.1) I don't see any mentioning of advantages of one of them over the other of them.
In our code base I saw that both are used.

Reply:
My own 2¢, but consistency, I guess? `-eq`, `-ne`, `-gt`, and `-lt` are all formatted as flags (similar to how bash uses flags to the `[` and `[[` commands to represent equality tests), such that `!` feels a bit out of place to me next to flag-style operators. In this case, though, changing from `!`/`-eq` to `-ne` also allows for removing a level of parentheses and making the code slightly more readable as a result.